Mathlib/Geometry/Manifold/SmoothManifoldWithCorners.lean

```diff
@@ -649,72 +646,127 @@ instance : ClosedUnderRestriction (contDiffGroupoid n I) :=
-    (I : ModelWithCorners 𝕜 E H) (M : Type*) [TopologicalSpace M] [ChartedSpace H M] extends
-    HasGroupoid M (contDiffGroupoid ∞ I) : Prop
+    (I : ModelWithCorners 𝕜 E H) (n : WithTop ℕ∞) (M : Type*)
+    [TopologicalSpace M] [ChartedSpace H M] extends
+    HasGroupoid M (contDiffGroupoid n I) : Prop
+
+@[deprecated (since := "2024-12-08")] alias SmoothManifoldWithCorners := IsManifold
 
-theorem SmoothManifoldWithCorners.mk' {𝕜 : Type*} [NontriviallyNormedField 𝕜] {E : Type*}
+/-- Building a `C^n` manifold from a `HasGroupoid` assumption. -/
+theorem IsManifold.mk' {𝕜 : Type*} [NontriviallyNormedField 𝕜] {E : Type*}
     [NormedAddCommGroup E] [NormedSpace 𝕜 E] {H : Type*} [TopologicalSpace H]
-    (I : ModelWithCorners 𝕜 E H) (M : Type*) [TopologicalSpace M] [ChartedSpace H M]
-    [gr : HasGroupoid M (contDiffGroupoid ∞ I)] : SmoothManifoldWithCorners I M :=
+    (I : ModelWithCorners 𝕜 E H) (n : WithTop ℕ∞)
+    (M : Type*) [TopologicalSpace M] [ChartedSpace H M]
+    [gr : HasGroupoid M (contDiffGroupoid n I)] : IsManifold I n M :=
   { gr with }
 
-theorem smoothManifoldWithCorners_of_contDiffOn {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+@[deprecated (since := "2024-12-08")] alias SmoothManifoldWithCorners.mk' := IsManifold.mk'
+
+theorem isManifold_of_contDiffOn {𝕜 : Type*} [NontriviallyNormedField 𝕜]
     {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E] {H : Type*} [TopologicalSpace H]
-    (I : ModelWithCorners 𝕜 E H) (M : Type*) [TopologicalSpace M] [ChartedSpace H M]
+    (I : ModelWithCorners 𝕜 E H) (n : WithTop ℕ∞) (M : Type*)
+    [TopologicalSpace M] [ChartedSpace H M]
     (h : ∀ e e' : PartialHomeomorph M H, e ∈ atlas H M → e' ∈ atlas H M →
-      ContDiffOn 𝕜 ∞ (I ∘ e.symm ≫ₕ e' ∘ I.symm) (I.symm ⁻¹' (e.symm ≫ₕ e').source ∩ range I)) :
-    SmoothManifoldWithCorners I M where
+      ContDiffOn 𝕜 n (I ∘ e.symm ≫ₕ e' ∘ I.symm) (I.symm ⁻¹' (e.symm ≫ₕ e').source ∩ range I)) :
+    IsManifold I n M where
   compatible := by
-    haveI : HasGroupoid M (contDiffGroupoid ∞ I) := hasGroupoid_of_pregroupoid _ (h _ _)
+    haveI : HasGroupoid M (contDiffGroupoid n I) := hasGroupoid_of_pregroupoid _ (h _ _)
     apply StructureGroupoid.compatible
 
-/-- For any model with corners, the model space is a smooth manifold -/
-instance model_space_smooth {𝕜 : Type*} [NontriviallyNormedField 𝕜] {E : Type*}
+@[deprecated (since := "2024-12-08")]
+alias smoothManifoldWithCorners_of_contDiffOn := isManifold_of_contDiffOn
+
+/-- For any model with corners, the model space is a `C^n` manifold -/
+instance intIsManifoldModelSpace {𝕜 : Type*} [NontriviallyNormedField 𝕜] {E : Type*}
     [NormedAddCommGroup E] [NormedSpace 𝕜 E] {H : Type*} [TopologicalSpace H]
-    {I : ModelWithCorners 𝕜 E H} : SmoothManifoldWithCorners I H :=
+    {I : ModelWithCorners 𝕜 E H} {n : WithTop ℕ∞} : IsManifold I n H :=
   { hasGroupoid_model_space _ _ with }
 
-end SmoothManifoldWithCorners
+end IsManifold
 
-namespace SmoothManifoldWithCorners
+namespace IsManifold
```

Comment:
Careful: can you add deprecation warnings for the renamed theorems in this section, please?

Reply:
As above, namespace changes shouldn't be deprecated in mathlib.